File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int n = nums.Length;
        int [] prefix = new int[n+1];
        int [] suffix = new int[n+1];
        prefix[0] = nums[0];
        suffix[n-1] = nums[n-1];
        for(int i = 1; i < n; i++) {
            prefix[i] = prefix[i-1] * nums[i];
        }
        for(int i = n-2; i >= 0; i--) {
            suffix[i] = suffix[i+1] * nums[i];
        }
        int[] ans = new int[n];

        ans[0] = suffix[1];
        ans[n-1] = prefix[n-2];

        for(int i = 1; i < n-1; i++) {
            ans[i] = prefix[i-1] * suffix[i+1];
        }

        return ans;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> freq;
        for(auto &it: nums) {
            freq[it]++;
        }
        vector<pair<int,int>> ans;
        for(auto [x,y]: freq) {
            ans.push_back({y,x});
        }
        sort(ans.rbegin(), ans.rend());
        vector<int> x;
        for(int i = 0; i < k; i++) {
            x.push_back(ans[i].second);
        }
        return x;
    }
};
